New package: CorbinDavenport.NexusTools version 5.7

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #246133

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

![Image](https://github.com/user-attachments/assets/dfe1ea5c-70b8-44a6-9f23-af9afe27a768)
![Image](https://github.com/user-attachments/assets/a06fef85-10de-4c5b-9206-b2a47e312fec)

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/246216)

Dragon1573 committed:
New package: CorbinDavenport.NexusTools version 5.7

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

stephengillie commented:
Automatic Validation ended with:
>  The FileSystemWatcher has detected an error System.IO.ErrorEventArgs

(Automated response - build 1037.)

rc-chuah commented:
> Automatic Validation ended with:
> 
> > The FileSystemWatcher has detected an error System.IO.ErrorEventArgs
> 
> (Automated response - build 1037.)

![triage-8-4-2025-capture](https://github.com/user-attachments/assets/c355814c-4c3b-47f3-ae50-14eddbfcb56b)

@stephengillie @Dragon1573 Need To Add Package Dependencies Package Identifier (ClockworkMod.UniversalADBDriver) In Installer Manifest Yaml File. If Not, It Will Generate A Validation-Unattended-Failed Error.

```yaml
Dependencies:
  PackageDependencies:
    - PackageIdentifier: ClockworkMod.UniversalADBDriver
```

Dragon1573 commented on manifests/c/CorbinDavenport/NexusTools/5.7/CorbinDavenport.NexusTools.installer.yaml:
```diff
@@ -0,0 +1,27 @@
+InstallerSwitches:
+  Silent: nexustools -i -w
+  SilentWithProgress: nexustools -i -w
+  Repair: nexustools -r -w
```
```suggestion
  Repair: nexustools -r -w
  Dependencies:
    PackageDependencies:
    - PackageIdentifier: ClockworkMod.UniversalADBDriver
```

- See <https://github.com/microsoft/winget-pkgs/pull/246216#issuecomment-2785824087>

Dragon1573 committed:
Apply suggestions from code review

- Add required dependencies of ADB Driver
- This can prevent unattended failure

Dragon1573 committed:
Update manifest: CorbinDavenport.NexusTools version 5.7

- Fix YAML indentation error.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
> [!IMPORTANT]
> 
> Unable to validate on my local _Windows Sandbox_ instance. I'm facing the "Handshake Exception". Can't sure if it's caused by my network environment (I'm behind a proxy provided by **Clash for Windows**, "UWP Loopback" enabled for Windows Sandbox).
> 
> ```text
> --> Installing WinGet
> --> Disabling safety warning when running installers
> Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.
> 
> --> Configuring Winget
> 已启用管理员设置 'LocalManifestFiles'。
> 已启用管理员设置 'LocalArchiveMalwareScanOverride'。
> 已启用管理员设置 'ProxyCommandLineOptions'。
> 将管理员设置 'DefaultProxy' 设置为 'http://alist.dragon1573.local:57890'。
> 
> --> Installing the Manifest 5.7
> 
> 已找到 Nexus Tools [CorbinDavenport.NexusTools] 版本 5.7
> 此应用程序由其所有者授权给你。
> Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
> 此包需要以下依赖项：
>   - 程序包
>       ClockworkMod.UniversalADBDriver
> (1/1) 已找到 Universal Adb Driver [ClockworkMod.UniversalADBDriver] 版本 1.0.4
> 此应用程序由其所有者授权给你。
> Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
> 正在下载 https://github.com/koush/adb.clockworkmod.com/releases/download/v1.0.0/UniversalAdbDriverSetup.msi
>   ██████████████████████████████  16.2 MB / 16.2 MB
> 已成功验证安装程序哈希
> 正在启动程序包安装...
> 已成功安装
> 
> 正在下载 https://github.com/corbindavenport/nexus-tools/releases/download/v5.7/nexustools-windows-x64.zip
>   ██████████████████████████████  2.40 MB / 2.40 MB
> 已成功验证安装程序哈希
> 正在提取存档...
> 已成功提取存档
> 正在启动程序包安装...
> 安装程序失败，退出代码为: 1
> 
> --> Refreshing environment variables
> 
> --> Comparing ARP Entries
> 
> DisplayName          DisplayVersion Publisher    ProductCode                            Scope
> -----------          -------------- ---------    -----------                            -----
> Universal Adb Driver 1.0.4          ClockworkMod {C0E08D8D-6076-4117-B644-2AF34F35B757} Machine
> 
> PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> C:\Users\WDAGUtilityAccount\AppData\Local\Temp\WinGet\CorbinDavenport.NexusTools.5.7\extracted\nexustools.exe -i -w
> [INFO] Nexus Tools 5.7 on AMD64
> [ OK ] Created folder at C:\Users\WDAGUtilityAccount\AppData\Roaming\NexusTools.
> [INFO] You agree to the Terms & Conditions by installing this software: https://developer.android.com/studio/terms
> [....] Downloading Platform Tools package, please wait.
> [EROR] There was an error downloading Platform Tools: HandshakeException: Handshake error in client (OS Error:
>         CERTIFICATE_VERIFY_FAILED: unable to get local issuer certificate(../../third_party/boringssl/src/ssl/handshake.cc:393))
> ```

Dragon1573 commented:
> [!IMPORTANT]
> 
> On my local host OS, similar issue occurs.
> 
> ```text
> 23:56:07 D:\...\winget-pkgs  [issues/246133 ≡ +0 ~1 -0 !] 13.838s pwsh$ gsudo { winget install -m .\manifests\c\CorbinDavenport\NexusTools\5.7\ }
> 已找到 Nexus Tools [CorbinDavenport.NexusTools] 版本 5.7
> 此应用程序由其所有者授权给你。
> Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
> 此包需要以下依赖项：
>   - 程序包
>       ClockworkMod.UniversalADBDriver
> (1/1) 已找到 Universal Adb Driver [ClockworkMod.UniversalADBDriver] 版本 1.0.4
> 此应用程序由其所有者授权给你。
> Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
> 正在下载 https://github.com/koush/adb.clockworkmod.com/releases/download/v1.0.0/UniversalAdbDriverSetup.msi
>   ██████████████████████████████  16.2 MB / 16.2 MB
> 已成功验证安装程序哈希
> 正在启动程序包安装...
> 已成功安装
> 
> 正在下载 https://github.com/corbindavenport/nexus-tools/releases/download/v5.7/nexustools-windows-x64.zip
>   ██████████████████████████████  2.40 MB / 2.40 MB
> 已成功验证安装程序哈希
> 正在提取存档...
> 已成功提取存档
> 正在启动程序包安装...
> 安装程序失败，退出代码为: 255
> 
> 23:57:17 D:\...\winget-pkgs  [issues/246133 ≡ +0 ~1 -0 !] 27.487s pwsh$ C:\Users\Dragon1573\AppData\Local\Temp\WinGet\CorbinDavenport.NexusTools.5.7\extracted\nexustools.exe -i -w
> [INFO] Nexus Tools 5.7 on AMD64
> [ OK ] Created folder at C:\Users\Dragon1573\AppData\Roaming\NexusTools.
> [INFO] You agree to the Terms & Conditions by installing this software: https://developer.android.com/studio/terms
> [....] Downloading Platform Tools package, please wait.
> Unhandled exception:
> Exception: The request failed because it wasn't able to reach the ipify service. This is most likely due to a networking error of some sort.
> #0      Ipify._send (package:dart_ipify/src/dart_ipify_base.dart:160)
> <asynchronous suspension>
> #1      Ipify.ipv4 (package:dart_ipify/src/dart_ipify_base.dart:54)
> <asynchronous suspension>
> #2      connectAnalytics (file:///d:/a/nexus-tools/nexus-tools/bin/main.dart:230)
> <asynchronous suspension>
> ```

rc-chuah commented:
@wingetbot run

rc-chuah commented:
> Important
> 
> On my local host OS, similar issue occurs.
> 
> ```
> 23:56:07 D:\...\winget-pkgs  [issues/246133 ≡ +0 ~1 -0 !] 13.838s pwsh$ gsudo { winget install -m .\manifests\c\CorbinDavenport\NexusTools\5.7\ }
> 已找到 Nexus Tools [CorbinDavenport.NexusTools] 版本 5.7
> 此应用程序由其所有者授权给你。
> Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
> 此包需要以下依赖项：
>   - 程序包
>       ClockworkMod.UniversalADBDriver
> (1/1) 已找到 Universal Adb Driver [ClockworkMod.UniversalADBDriver] 版本 1.0.4
> 此应用程序由其所有者授权给你。
> Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
> 正在下载 https://github.com/koush/adb.clockworkmod.com/releases/download/v1.0.0/UniversalAdbDriverSetup.msi
>   ██████████████████████████████  16.2 MB / 16.2 MB
> 已成功验证安装程序哈希
> 正在启动程序包安装...
> 已成功安装
> 
> 正在下载 https://github.com/corbindavenport/nexus-tools/releases/download/v5.7/nexustools-windows-x64.zip
>   ██████████████████████████████  2.40 MB / 2.40 MB
> 已成功验证安装程序哈希
> 正在提取存档...
> 已成功提取存档
> 正在启动程序包安装...
> 安装程序失败，退出代码为: 255
> 
> 23:57:17 D:\...\winget-pkgs  [issues/246133 ≡ +0 ~1 -0 !] 27.487s pwsh$ C:\Users\Dragon1573\AppData\Local\Temp\WinGet\CorbinDavenport.NexusTools.5.7\extracted\nexustools.exe -i -w
> [INFO] Nexus Tools 5.7 on AMD64
> [ OK ] Created folder at C:\Users\Dragon1573\AppData\Roaming\NexusTools.
> [INFO] You agree to the Terms & Conditions by installing this software: https://developer.android.com/studio/terms
> [....] Downloading Platform Tools package, please wait.
> Unhandled exception:
> Exception: The request failed because it wasn't able to reach the ipify service. This is most likely due to a networking error of some sort.
> #0      Ipify._send (package:dart_ipify/src/dart_ipify_base.dart:160)
> <asynchronous suspension>
> #1      Ipify.ipv4 (package:dart_ipify/src/dart_ipify_base.dart:54)
> <asynchronous suspension>
> #2      connectAnalytics (file:///d:/a/nexus-tools/nexus-tools/bin/main.dart:230)
> <asynchronous suspension>
> ```

@Dragon1573 It Seems Like There's A Problem Domain Name Server (DNS), It Can Be Due To Internet Service Provider (ISP) Or The Issue From Domain Name Server (DNS) Itself And It Also Can Be Due To A Networking Issue.

The Workaround Is Try To Run The Validation Pipeline Again And See If There's Any Validation-Installation-Error. If Yes Then Try Running Nexus Tools Without Plausible Analytics With The Flag `nexustools -i -w --no-analytics` Because It's Due To The `connectAnalytics()` Function Trying To Connect To Ipify Service To Get The IP Address To Include In The Analytics But Failed And If Running Nexus Tools Without Plausible Analytics Works Then Include The `--no-analytics` Flag In The Installer Manifest Yaml File Like The Example Below.

```yaml
InstallerSwitches:
  Silent: nexustools -i -w --no-analytics
  SilentWithProgress: nexustools -i -w --no-analytics
  Repair: nexustools -r -w
```

Dragon1573 commented:
> It seems like there's a problem with the Domain Name Server (DNS). It can be due to the Internet Service Provider (ISP) or the issue from the Domain Name Server (DNS) itself, and it also can be due to a networking issue.
>
> The workaround is to try running the validation pipeline again and see if there's any <https://github.com/microsoft/winget-pkgs/labels/Validation-Installation-Error>. If yes, then try running Nexus Tools without Plausible Analytics with the flag `nexustools -i -w --no-analytics` because it's due to the `connectAnalytics()` function trying to connect to Ipify service to get the IP address to include in the analytics but failed. If running Nexus Tools without Plausible Analytics works, then include the `--no-analytics` flag in the installer manifest YAML file like the example below.
> 
> ```yaml
> InstallerSwitches:
>   Silent: nexustools -i -w --no-analytics
>   SilentWithProgress: nexustools -i -w --no-analytics
>   Repair: nexustools -r -w
> ```

✅ After manually configure proxy in Windows Sandbox instance, manual reinstall success.

![image](https://github.com/user-attachments/assets/8ffe3354-f7e0-40a6-b247-59efd7828f21)

Dragon1573 committed:
Update manifest: CorbinDavenport.NexusTools version 5.7

- Add `--no-analytics` to silent install switches, preventing unnecessary telemetry data collection.
- May be able to bypass the network issue on some device.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>
